primitives/api/src/lib.rs

```diff
@@ -344,6 +344,54 @@ pub use sp_api_proc_macro::decl_runtime_apis;
 /// ```
 /// In this case `Balance` api version 3 is being implemented for `Runtime`. The `impl` block
 /// must contain all methods declared in version 3 and below.
+///
+/// # Conditional version implementation
+///
+/// `decl_runtime_apis!` supports `cfg_attr` attribute for conditional compilation. For example
+/// let's say you want to implement a staging version of the runtime api and put it behind a
+/// feature flag. You can do it this way:
+/// ```ignore
+/// pub struct Runtime {}
+/// sp_api::decl_runtime_apis! {
+///     pub trait ApiWithStagingMethod {
+///         fn stable_one(data: u64);
+///
+///         #[api_version(99)]
+///         fn staging_one();
+///     }
+/// }
+///
+/// sp_api::impl_runtime_apis! {
+///     #[cfg_attr(feature = "enable-staging-api", api_version(99))]
+///     impl self::ApiWithStagingMethod<Block> for Runtime {
+///         fn stable_one(_: u64) {}
+///
+///         #[cfg_attr(feature = "enable-staging-api", api_version(99))]
```

Comment:
Why isn't this just `#[cfg(feature = "enable-staging-api")]`? 

Why do we need to provide the version here again?

Reply:
The idea is to either implement version 99 of the api (let's call it staging for brevity) or some other stable version (1 in this case) depending on a supplied feature flag.
So if the feature is enabled, we are building a staging version which is supposed to add additional methods ontop the stable ones. In the impl block the staging methods also have got `cfg_attr` so that we can discriminate them.

Does this make sense?

Reply:
I understand now. You are right, it will be better your way. I'll rework it.

Reply:
Fixed